### ISSUE TITLE
Set ConfigureAwait to false all the way down

### ISSUE THREAD

**Issue by giuliohome** [linq2db/linq2db#1366]
I need to use `ConfigureAwait(false)` in a business `WebMethod` and I can't implement the DB retrieval with linq2db due to this issue.

See [note](https://blog.stephencleary.com/2012/07/dont-block-on-async-code.html#preventing-the-deadlock): "You would have to use ConfigureAwait(false) for every await in the transitive closure of all methods called by the blocking code, including all third- and second-party code."

Either you set `ConfigureAwait(false)` in the internals of your async functions (which has to be done in the `AsyncExtensions.cs`, `DataReaderAsync.cs`, `DataConnection.Async.cs` and so on) or you expose this setting to the lin2db users of those async functions (if someone wants it to be configurable).

**Comment by sdanyliv** [comment]
@giuliohome, it is not obvious.
You can check also second answer here: https://stackoverflow.com/questions/13489065/best-practice-to-call-configureawait-for-all-server-side-code 

Internally we don't like such approach. Which may make ConfigureAwait mess in our code.

**Comment by giuliohome** [comment]
@sdanyliv 
First of all, I want to be sure we are on the same page, notice that you can directly link to a specific answer (using the "share" url) and not to the question. So is the second answer [this one](https://stackoverflow.com/a/13489639) "Brief answer to your question: No. bla bla bla..."?

I think so. Now, that answer has a "TL;DR version of the long answer" which I'm going to quickly read.

"If you are writing a library where you don't know your consumer and don't need a synchronization context (which you shouldn't in a library I believe), you should always use ConfigureAwait(false). Otherwise, the consumers of your library may face deadlocks by consuming your asynchronous methods in a blocking fashion."

In conclusion, this answer of tugberk on SO is saying - IMO - that my issue request is correct whilst the implementation of linq2db is wrong... This is my interpretation of such answer and I think that the answer is correct as far as I correctly interpret it.

**Comment by sdanyliv** [comment]
@giuliohome, we will investigate. We currently analyzing Pros and Cons.

**Comment by giuliohome** [comment]
I think I've found the issue
it is the blocking `async:false` in the ajax call of the jtable implementation [here]
(https://github.com/volosoft/jtable/blob/master/jquery.jtable.js#L915) 
I guess I have to wrap the c# async webmethod with a correct ajax myself or move the issue there
